sortmerna: init at 3.0.3

luispedro opened the pull request:
This is a useful package in bioinformatics, the corresponding bioconda package has [> 50,000 downloads](https://anaconda.org/bioconda/sortmerna)


###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested execution of all binary files (usually in `./result/bin/`)

luispedro committed:
sortmerna: init at 3.0.3

This is a useful package in bioinformatics, namely in processing
short-read sequencing data from metagenomes

tadeokondrak commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,32 @@
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "biocore";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

tadeokondrak commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  meta = with stdenv.lib; {
+    description = "SortMeRNA is a program tool for filtering, mapping and OTU-picking NGS reads in metatranscriptomic and metagenomic data.";
+    license = licenses.lgpl;
```
```suggestion
    license = licenses.lgpl3;
```

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  # The `make install` command shipped with the package installs header files,
+  # which are not necessary:
+  installPhase = ''
```
I think it would be cleaner to just use the default `installPhase` and just remove the headers in `postFixup`?

luispedro commented:
I had started from the bioconda package for this, but I realized that there is a newer release, but I need https://github.com/NixOS/nixpkgs/pull/64987 to be committed for the new version to compile on Nix

worldofpeace commented:
@luispedro Request granted :sparkles:

luispedro commented:
@worldofpeace : much appreciated. I updated the package and it compiles cleanly on top of master. Some of the previously raised issues have gone away in the new release (the headers are no longer installed by default), the remaining I fixed manually.

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,41 @@
+    sha256 = "0zx5fbzyr8wdr0zwphp8hhcn1xz43s5lg2ag4py5sv0pv5l1jh76";
+  };
+
+  patches = [ ./CMakeInstallPrefix.patch ];
```
This was commented out in a commit upstream https://github.com/biocore/sortmerna/commit/4d36d620a3207e26cf3f588d4ec39889ea21eb79.

Can you `fetchpatch` it?
> https://nixos.org/nixpkgs/manual/#sec-patches

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,41 @@
+  preConfigure = ''
+    cmakeFlagsArray=(
+      $cmakeFlagsArray
+      "-DCMAKE_INSTALL_PREFIX=$out")
```
I did not need this for things to install correctly when testing locally.
Please remove.

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+
+  meta = with stdenv.lib; {
+    description = "SortMeRNA is a program tool for filtering, mapping and OTU-picking NGS reads in metatranscriptomic and metagenomic data.";
```
```suggestion
    description = "next-generation sequence filtering and alignment tool";
```

We shouldn't have periods at the end of `description` or restate the package name.
Description is from https://github.com/biocore/sortmerna

worldofpeace commented on pkgs/applications/science/biology/sortmerna/CMakeInstallPrefix.patch:
```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt	2019-07-17 09:37:47.964515902 +0200
```
Be sure to delete this file

luispedro commented:
Thanks. Forced push new version.

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,40 @@
+    "-DRAPIDJSON_HOME=${rapidjson}"
+  ];
+
+
```
drop trailing whitespace

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+  patches = [
+    (fetchpatch {
+      name="CMakeInstallPrefix.patch";
```
whitespace like for `sha256`

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,40 @@
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "biocore";
+    rev = version;
```
```suggestion
    rev = "v${version}";
```

I'm seeing tags formatted in this way
* https://github.com/biocore/sortmerna/tree/v3.0.3

worldofpeace commented on pkgs/applications/science/biology/sortmerna/default.nix:
```diff
@@ -0,0 +1,38 @@
+  patches = [
+    (fetchpatch {
+      name = "CMakeInstallPrefix.patch";
+      url="https://github.com/biocore/sortmerna/commit/4d36d620a3207e26cf3f588d4ec39889ea21eb79.patch";
```
```suggestion
      url = "https://github.com/biocore/sortmerna/commit/4d36d620a3207e26cf3f588d4ec39889ea21eb79.patch";
```

luispedro replied:
Thanks! fixed.

Ekleog commented:
@GrahamcOfBorg build sortmerna

Ekleog commented:
Looks like `sortmerna` doesn't build on aarch64-linux. Maybe either somehow fix that (not sure whether it's supposed to) or mark it as disabled by having eg. `meta.platforms = "x86_64-linux"`?

luispedro commented:
Seeing the error and the code, it seems that it relies on using intrinsics, so it's tied to x86

Ekleog commented:
@GrahamcOfBorg build sortmerna

(checking whether it builds with `x86_64-darwin`, as the switch from `linux` to `x86_64` likely enabled it)

Ekleog commented:
Thank you!
